dojo_plugin/api/v1/registry.py

```diff
@@ -0,0 +1,83 @@
+from flask import request
+from flask_restx import Namespace, Resource
+
+import re
+
+from ...config import REGISTRY_API_SECRET
+from ...models import Dojos, DojoAdmins
+from CTFd.models import Users, Admins
+from CTFd.utils.crypto import verify_password
+
+
+registry_namespace = Namespace(
+    "registry", description="Endpoint to support registry auth checks"
+)
+
+
+def auth_check(authorization):
+    if not authorization or not authorization.startswith("Bearer "):
+        return {"success": False, "error": "Unauthorized"}, 401
+
+    token = authorization.split(" ")[1]
+    if not (REGISTRY_API_SECRET and token == REGISTRY_API_SECRET):
+        return {"success": False, "error": "Unauthorized"}, 401
+
+    return None, None
+
+
+@registry_namespace.route("/verify")
+class RegistryVerify(Resource):
+    def post(self):
+        authorization = request.headers.get("Authorization")
+        res, code = auth_check(authorization)
+        if res:
+            return res, code
+
+        data = request.get_json() or {}
+        username = data.get("username")
+        password = data.get("password")
+        repository = data.get("repository")
+        actions = data.get("actions") or []
+
+        if not username or not password:
+            return {"success": False, "error": "Missing credentials"}, 400
+
+        user = Users.query.filter((Users.name == username) | (Users.email == username)).first()
+        if not user or not verify_password(password, user.password):
+            return {"success": False, "error": "Invalid credentials"}, 401
+
+
+        is_global_admin = Admins.query.filter_by(id=user.id).first() is not None
+
+        if not repository:
+            return {"success": True}
+
+
+        m = re.match(r"^(?P<dojo_id>[a-z0-9][a-z0-9-]*)-(?P<dojo_hex>[0-9a-f]{8})(?:/.*)?$", repository)
+        if not m:
+            return {
+                "success": False,
+                "error": "Invalid repository format. Reference ID looks like '<dojo_name>~<dojoid>' Check the admin page; use this format '<dojo_name>-<dojoid>/yourimage' as the repo name"
+            }, 403
+
+        dojo_name = m.group("dojo_id")
+        dojo_hex = m.group("dojo_hex")
```

Comment:
All dojos do not have a `dojo_hex`.  A dojo can be made "official" at which point the dojo_hex is removed from the reference_id.